newIDE/app/src/ExpressionAutocompletion/index.js

```diff
@@ -292,6 +294,59 @@ const getAutocompletionsForText = function(
   }));
 };
 
+const getAutocompletionsForVariable = function(
+  expressionAutocompletionContext: ExpressionAutocompletionContext,
+  completionDescription: gdExpressionCompletionDescription
+): Array<ExpressionAutocompletion> {
+  const prefix: string = completionDescription.getPrefix();
+  const type: string = completionDescription.getType();
+  const objectName: string = completionDescription.getObjectName();
+  const { project, scope } = expressionAutocompletionContext;
+  const layout = scope.layout;
+
+  let variablesContainer: gdVariablesContainer;
+  if (type === 'globalvar') {
+    if (!project) {
+      // No variable completion
+      return [];
+    }
+    variablesContainer = project.getVariables();
+  } else if (type === 'scenevar') {
+    if (!layout) {
+      // No variable completion
+      return [];
+    }
+    variablesContainer = layout.getVariables();
+  } else if (type === 'objectvar') {
+    const object = getObjectByName(project, layout, objectName);
+    if (!object) {
+      // No variable completion for unknown objet
+      return [];
+    }
+    variablesContainer = object.getVariables();
+  }
+
+  const definedVariableNames = enumerateVariables(variablesContainer)
+    .map(({ name, isValidName }) =>
+      isValidName
+        ? name
+        : // Hide invalid variable names - they would not
+          // be parsed correctly anyway.
+          null
+    )
+    .filter(Boolean);
+
+  const filteredVariablesList = filterStringList(definedVariableNames, prefix);
+
+  return filteredVariablesList.map(variableName => ({
+    kind: 'Text',
```

Comment:
nitpicking: a variable name is technically not a "Text" (which means, in this context, a "string literal", i.e: a sequence of characters enclosed in double quotation marks), but an "Identifier" (which means, in this context, a sequence of characters, excepted for special separator characters (like spaces, comma, dots...) and that does not start with a number).

So the autocompletion here if really providing you with an "Identifier", so we should set the kind to it (not that it will change anything, but something in the future might rely on this information).

Also add it to the `export type ExpressionAutocompletion` type (like Text) and this should be good to go!